data_safe_haven/infrastructure/project_manager.py

```diff
@@ -309,9 +310,13 @@ def install_plugins(self, workspace: automation.Workspace) -> None:
             raise DataSafeHavenPulumiError(msg) from exc
 
     def log_exception(self, exc: automation.CommandError) -> None:
-        with suppress(IndexError):
-            stderr = str(exc).split("\n")[3].replace(" stderr: ", "")
-            self.log_message(f"Pulumi output: {stderr}")
+        for error_line in str(exc).split("\n"):
+            if any(word in error_line for word in ["error:", "stderr:"]):
+                message = strip_ansi(
+                    error_line.replace("error:", "").replace("stderr:", "")
+                ).strip()
```

Comment:
I'm not sure why the str -> Text -> str -> Text is necessary, but otherwise the changes here are good.

Could this bit be moved to #2113 while we figure that out?

Reply:
It's because you quite often get Pulumi output that looks like this:

`"\033[31;merror:\033[0m Code='AnErrorCode'"`

<img width="383" alt="Screenshot 2024-08-09 at 10 48 39" src="https://github.com/user-attachments/assets/7e1752b7-9308-4ac1-a2e2-3e6daf405ff0">

which we'd like to log as `"Pulumi output: Code='AnErrorCode'"`.

If we try to do `error_line.replace("error:", "").strip()` then the first whitespace character is masked by the ANSI colour codes so you get:

`"Pulumi output: \033[31;m\033[0m Code='AnErrorCode'"`.

which looks like

<img width="450" alt="Screenshot 2024-08-09 at 10 48 56" src="https://github.com/user-attachments/assets/5d0e7568-f283-41ae-97a6-f7c4b02f9034">

and effectively a double-space in the output.

If we can strip the ANSI codes first, then the `strip()` call will work. There are other ways to strip ANSI codes but they mainly involve complicated regexes. If you'd prefer to do this then we can.

I'll move this commit to the other PR though.